R/extract_fin_year.R

```diff
@@ -29,18 +29,14 @@ extract_fin_year <- function(date) {
   # and then match them back on to the original input. This vastly improves
   # performance for large inputs.
 
-  unique_date <- unique(date)
-
-  unique_fy_q <-
-    lubridate::year(unique_date) - (lubridate::month(unique_date) %in% 1:3)
-
-  unique_fy <- ifelse(
-    is.na(unique_date),
-    NA_character_,
-    paste0(unique_fy_q, "/", (unique_fy_q %% 100L) + 1L)
-  )
-
-  fin_years <- unique_fy[match(date, unique_date)]
-
-  return(fin_years)
+  # Note: lubridate year and month coerce to double
+  # We only need integers for our purposes
+  posix <- as.POSIXlt(date, tz = lubridate::tz(date))
+  y <- posix$year + 1900L
+  m <- posix$mon + 1L
+  fy <- y - ( (m - 3L) %/% 1L <= 0L )
```

Comment:
The `%/% 1` here isn't doing anything.

I think the simplest (and fastest) way to do this is to just check if `m` is less than 3.
```suggestion
  fy <- y - (m <= 3L)
```

Reply:
Agree, I've removed the unnecessary operations.